### PR TITLE
Reference pressure in diagnostic coordinates

### PR DESCRIPTION
**Pull request by ashao**
Support for users changing the reference pressure for state-dependent coordinates (e.g. HYCOM-like and continuous isopycnal) was only supported for the main, prognostic coordinate. If users requested the continuous isopycnal coordinate, the reference pressure would always be set to 2000dbar. This PR updates the run parameters so that users can also set the reference pressure for diagnostic coordinates (i.e. sigma-0, sigma-2) and similar options using the runtime.

### CODE REVIEW

File: src/ALE/coord_hycom.F90 [hunk 1]
```diff
@@ -69,11 +69,13 @@ subroutine end_coord_hycom(CS)
 end subroutine end_coord_hycom
 
 !> This subroutine can be used to set the parameters for the coord_hycom module
-subroutine set_hycom_params(CS, max_interface_depths, max_layer_thickness, interp_CS)
+subroutine set_hycom_params(CS, max_interface_depths, max_layer_thickness, interp_CS, ref_pressure)
```
Comment (adcroft) [inline src/ALE/coord_hycom.F90:72]: This new argument is not used. It needs a
```if (present(ref_pressure)) CS%ref_pressure = ref_pressure```
added

Comment (ashao) [reply src/ALE/coord_hycom.F90:72]: Thanks @adcroft

Comment (adcroft) [reply src/ALE/coord_hycom.F90:72]: Looks like it might be easier to not add this new arg?

Comment (adcroft) [reply src/ALE/coord_hycom.F90:72]: Given that the hycom CS doesn't have a ref_pressure to change, perhaps easier to drop this new argument. The PR is asimed at flexible diagnostics and the hybrid coordinates should not be used for diagnostics.

Comment (ashao) [reply src/ALE/coord_hycom.F90:72]: That's a fair argument. Let me do a more careful compilation test this time -_-. Serves me right for trying to submit a PR before vacation.

**Review by adcroft — APPROVED** [review]

**Comment by marshallward** [comment]
Gaea regression: https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/17766 :heavy_check_mark: :yellow_circle:

**Comment by adcroft** [comment]
Expected new parameters are `DIAG_COORD_P_REF_RHO2` and `DIAG_COORD_REGRID_COMPRESSIBILITY_FRACTION_RHO2` which seems consistent with pipeline results

**Comment by adcroft** [comment]
Passed testing: https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/17766

**Comment by ashao** [comment]
Thanks @adcroft and @marshallward

**Pull request merged**